src/averageAges.js

```diff
@@ -55,7 +65,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const parents = people.filter(person => (onlyWithSon)
+    ? person.sex === 'm' && (
+      people.find(mother => person.mother === mother.name))
+    : people.find(mother => person.mother === mother.name)
+  );
+
+  function getMotherByYear(param) {
+    return people.find(mother => param === mother.name);
+  }
+
+  const diffAge = parents.map(person => person.born
+    - getMotherByYear(person.mother).born);
```

Comment:
you can get mother born directly from function, but ok
```suggestion
  function getMotherByYear(param) {
    return people.find(mother => param === mother.name).born;
  }

  const diffAge = parents.map(person => person.born
    - getMotherByYear(person.mother));
```
smth like this